server/src/main/java/com/objectcomputing/checkins/services/slack/kudos/KudosChannelReader.java

```diff
@@ -0,0 +1,57 @@
+package com.objectcomputing.checkins.services.slack.kudos;
+
+import com.objectcomputing.checkins.services.slack.SlackReader;
+import com.objectcomputing.checkins.configuration.CheckInsConfiguration;
+
+import com.slack.api.model.Message;
+
+import io.micronaut.scheduling.annotation.Scheduled;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jakarta.inject.Singleton;
+import jakarta.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.time.LocalDateTime;
+
+@Singleton
+public class KudosChannelReader {
+    private static final Logger LOG = LoggerFactory.getLogger(KudosChannelReader.class);
+
+    @Inject
+    private KudosChannelReadTimeStore kudosChannelReadTimeStore;
+
+    @Inject
+    private CheckInsConfiguration configuration;
+
+    @Inject
+    private SlackReader slackReader;
+
+    @Inject
+    private SlackKudosCreator slackKudosCreator;
+
+    @Scheduled(fixedDelay = "10m")
```

Comment:
I don't think we can count on this because sometimes we scale down completely...and we will end up with multiple channel readers when the app is busy as well. We can end up with up to 10 instances running currently, I think.

Reply:
@mkimberlin I just checked in a change to remove the scheduled task and tie it to the `CheckServicesController`.